src/scenes/AcceleratorRouter/Modules/EventsTable.tsx

```diff
@@ -71,48 +71,77 @@ export default function EventsTable(props: EventsTableProps): JSX.Element {
   const [selectedRows, setSelectedRows] = useState<ModuleEventPartial[]>([]);
   const [eventToEdit, setEventToEdit] = useState<ModuleEventPartial>();
 
+  const generatedIds = new Set();
+
+  function generateUniqueId() {
+    let newId: number;
+    do {
+      newId = Math.random();
+    } while (generatedIds.has(newId)); // Check for uniqueness
+
+    generatedIds.add(newId);
+    return newId;
+  }
+
   const onAddEvent = (eventToAdd: ModuleEventPartial) => {
+    let newEventToAdd = { ...eventToAdd };
+    if (eventToAdd.id === -1 && !eventToAdd.feId) {
+      newEventToAdd = { ...eventToAdd, feId: generateUniqueId() };
```

Comment:
You can you also use `Symbol()` here. [Docs here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol). Basically, it is always guaranteed unique